components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,666 @@
+can have optional requirements which are set by ``->setCategoryRequirement()`` method.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\ShippingCategory;
+        use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+
+        $shippingCategory = new ShippingCategory();
+        $shippingCategory->setName('Regular'); // Regular weight items
+
+
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setCategory($shippingCategory); //default null, detach
+        $shippingMethod->getCategory(); // Output will be ShippingCategory object
+        $shippingMethod->setCategory();
+        $shippingMethod->getCategory(); // Output will be null
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ALL);
+        // the shipping method is now support any shipment (or shipping subject) which contains
+        // at least one shippable with the same category. If shipment have many shippable object,
+        // all of them have to have the same category.
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_NONE);
+        // if none shippables (in shipment) have reference the same category as shippingMethod,
+        // the shippingMethod is allowed for this shipment
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ANY);
+        // if one  of the shippables in shipping have the same category as shippingMethod,
+        // the shipping Method is allowed for this shipment
+
+.. note::
+    If you what to understand how category requirements influence on relation between shipping method and shipment go to
+    :ref:`method-resolver`.
+
+Setting Rule
+~~~~~~~~~~~~
+
+Every shipping method can have many rules which defines additional requirements for shipment. If shipment do not
+fulfil this requirements (for example rule sets expected quantity of shipment items to 2, but shipment has only one **ShipmentItem**)
+then shippingMethod with this rule cannot be used to this shipment.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+
+        $shippingMethod->addRule($rule);
+        $shippingMethod->hasRule($rule); // returns true
+        $shippingMethod->getRules(); // collection of rules with count equals 1
+        $shippingMethod->removeRule($rule);
+        $shippingMethod->hasRule($rule); // returns false
+
+Shipping Method Translation
+---------------------------
+
+**ShippingMethodTranslation** allows translate name of shipping method according to the given locales. To see how to use translation
+please go to :ref:`component_translation_basic-translations`.
```

Comment:
* `**ShippingMethodTranslation** allows shipping method's name translation according to given locales.`  :)